internal/scheduler/submitcheck.go

```diff
@@ -100,10 +103,16 @@ func (srv *SubmitChecker) updateExecutors(ctx context.Context) {
 			}
 			srv.mu.Unlock()
 			if err != nil {
-				log.WithError(err).Errorf("Error constructing node db for executor %s", executor.Id)
+				log.
+					WithError(err).
```

Comment:
Why not use the `WithStacktrace` method?

Reply:
The stacktrace method only considers the first error.  I there is a wrapped error and the stacktrace on the top-level error is missing, but present on the cause, the stack is lost.

Reply:
I don't think that's supposed to me true.

Reply:
Actually I think I might have misunderstood what you were saying. If you are asking why this isn't using `logging.withStacktrace`, this does now get stacktraces from causes (although it didn't before).  The reason is that the this allows you to use the builder pattern throughout:
```
log.
WithField(logging.Stacktrace, logging.ExtractStack(err)).
WithError(err).
WithField(foo)
```
rather than
```
logging.WithStacktrace(
log.
WithField(logging.Stacktrace, logging.ExtractStack(err)).
WithError(err).
WithField(foo)
)
```

Reply:
I prefer this
```
logging.WithStacktrace(log, err).WithField(foo)
```